pkg/workflows/dontime/request.go

```diff
@@ -0,0 +1,66 @@
+package dontime
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+type Request struct {
+	ExpiresAt time.Time
+
+	// CallbackCh is a channel to send a response back to the requester
+	// after the request has been processed or timed out.
+	CallbackCh chan DonTimeResponse
+
+	WorkflowExecutionID string
+	SeqNum              int
+}
+
+func (r *Request) ID() string {
+	return r.WorkflowExecutionID
+}
+
+func (r *Request) ExpiryTime() time.Time {
+	return r.ExpiresAt
+}
+
+func (r *Request) SendResponse(ctx context.Context, resp DonTimeResponse) {
+	select {
+	case <-ctx.Done():
+		return
+	case r.CallbackCh <- resp:
+		close(r.CallbackCh)
+	}
+}
+
+func (r *Request) SendTimeout(ctx context.Context) {
```

Comment:
Can we make sure that SendTimeout and SendResponse are non-blocking and remove context from them? We generally shouldn't be calling anything blocking from inside OCR plugin. In this case we simply need to make the channel of size 1 (which it already is) and make sure that we don't respond more than once (which I'm almost sure we do...?).